### PR TITLE
fix(persons): properly update and unset properties

### PR DESCRIPTION
**Pull request by macobo**
## Problem

Solves https://github.com/PostHog/posthog/issues/9856, supercedes https://github.com/PostHog/posthog/pull/9855

## Changes

Person property updating/deleting caused postgres and clickhouse to go out of sync. This is now fixed.

A gotcha here is that this updating happens asynchronously now from the request.

### CODE REVIEW

File: frontend/src/scenes/persons/personsLogic.ts [hunk 1]
```diff
@@ -188,6 +190,21 @@ export const personsLogic = kea<personsLogicType>({
...
+            const person = values.person
+
+            if (person) {
+                const updatedProperties = { ...person.properties }
+                delete updatedProperties[key]
+
+                actions.setPerson({ ...person, properties: updatedProperties })
+                await api.create(`api/person/${person.id}/delete_property`, { $unset: [key] })
+
+                // :TODO: Toast!
```
Comment (yakkomajuri) [inline frontend/src/scenes/persons/personsLogic.ts:203]: should be a quick thing - suppose we don't need a follow up PR for it?

File: posthog/api/person.py [hunk 2]
```diff
@@ -101,6 +103,26 @@ class Meta:
...
+            team_id=instance.team_id,
+            now=datetime.now(),
+            sent_at=None,
+            event={
+                "event": "$set",
+                "properties": {"$set": validated_data["properties"]},
+                "distinct_id": instance.distinct_ids[0],
+                "timestamp": datetime.now().isoformat(),
+            },
+        )
```
Comment (yakkomajuri) [inline posthog/api/person.py:124]: should we `log_activity` here?

not too familiar with this whole thing but yeah

**Review by yakkomajuri — APPROVED** [review]
thanks for taking this on

File: posthog/api/person.py (hunk 2 above)
Comment (macobo) [reply posthog/api/person.py:124]: This seems to be already handled according to tests.

File: frontend/src/scenes/persons/personsLogic.ts (hunk 1 above)
Comment (yakkomajuri) [reply frontend/src/scenes/persons/personsLogic.ts:203]: "Property deletion request issued successfully" or "Property deletion triggered successfully"?

and then if the request errors let's `setPerson` back to just `person` and toast that the request errored

Comment (yakkomajuri) [reply frontend/src/scenes/persons/personsLogic.ts:203]: Given the UI updates also happy to just toast + update on error

Comment (macobo) [reply frontend/src/scenes/persons/personsLogic.ts]: AFAIK error handling is already done outside of logics, so adding a success message.

**Comment by macobo** [comment]
Updated the code - it was _still_ sneakily updating Persons behind closed doors.

cc @pauldambra I needed to touch the activity logging code here a bit. Let me know if you see any problems with that!

**Review by yakkomajuri — APPROVED** [review]
👍 from me - I was actually looking for the old update method's deletion before but couldn't find it, was hidden under lifecycle there

File: posthog/models/activity_logging/activity_log.py [hunk 3]
```diff
@@ -148,9 +148,10 @@ def log_activity(
     scope: str,
     activity: str,
     detail: Detail,
+    force_save: bool = False,
 ) -> None:
     try:
-        if activity == "updated" and (detail.changes is None or len(detail.changes) == 0):
+        if activity == "updated" and (detail.changes is None or len(detail.changes) == 0) and not force_save:
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/models/activity_logging/activity_log.py:154]: Yep, that's what we came up with after a quick call but it's not great for the front end 

I've proposed #9991 instead... CI is failing on that but for unrelated reasons.

It's only a single commit different to this branch

**Pull request merged**